### PR TITLE
Moved types of app components to separate files

### LINKED ISSUE #74: Move types in separate file for app components
Provided examples: 
image
image

### PR DESCRIPTION
**Pull request by ValKoDim**
Moved the types for app components to separate files within the folders of each of them.

### CODE REVIEW

File: code/src/components/app-container/AppContainer.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import Container, { ContainerProps } from "@mui/material/Container";
-
-export type AppContainerProps = ContainerProps;
+import Container from "@mui/material/Container";
+import { AppContainerProps } from "./AppContainer.types";
```
Comment (D1nviz) [inline code/src/components/app-container/AppContainer.tsx]: use alias import here

File: code/src/layouts/page-wrapper/PageWrapper.tsx [hunk 2]
```diff
@@ -1,8 +1,7 @@
 import { Ref, forwardRef, PropsWithChildren } from "react";
 
-import AppContainer, {
-  AppContainerProps
-} from "@/components/app-container/AppContainer";
+import AppContainer from "@/components/app-container/AppContainer";
+import { AppContainerProps } from "@/components/app-container/AppContainer.types";
 
 import "./PageWrapper.scss";
```
Comment (D1nviz) [inline code/src/layouts/page-wrapper/PageWrapper.tsx]: use alias here

File: code/src/components/app-button/AppButton.tsx [hunk 3]
```diff
@@ -1,25 +1,10 @@
-import Button, { ButtonProps } from "@mui/material/Button";
+import Button from "@mui/material/Button";
 import AppLoader from "@/components/app-loader/AppLoader";
 import cn from "@/utils/cn";
+import { AppButtonProps } from "./AppButton.types";
```
Comment (mhevyk) [inline code/src/components/app-button/AppButton.tsx]: Use alias here

File: code/src/components/app-icon-button/AppIconButton.tsx [hunk 4]
```diff
@@ -1,10 +1,5 @@
-import { LinkProps as ReactRouterDomLinkProps } from "react-router-dom";
-import { LinkProps as MuiLinkProps } from "@mui/material/Link";
-import IconButton, { IconButtonProps } from "@mui/material/IconButton";
-
-type AppIconButtonProps = IconButtonProps &
-  MuiLinkProps &
-  Partial<ReactRouterDomLinkProps>;
+import IconButton from "@mui/material/IconButton";
+import { AppIconButtonProps } from "./AppIconButton.types";
```
Comment (mhevyk) [inline code/src/components/app-icon-button/AppIconButton.tsx]: Use alias

File: code/src/components/app-logo/AppLogo.types.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { ComponentPropsWithoutRef } from "react";
+
+export const COMPONENT = "img";
```
Comment (mhevyk) [inline code/src/components/app-logo/AppLogo.types.ts]: It's not a type, better keep it local for `AppLogo`

Comment (ValKoDim) [reply code/src/components/app-logo/AppLogo.types.ts]: It is also used in AppLogo.tsx as a prop for the AppBox component. I just thought it is redundant to create it a second time there, so i just decided to import it.

Comment (mhevyk) [reply code/src/components/app-logo/AppLogo.types.ts]: Constant should not be stored in types, it is real javascript entity, not typescript. Of course we can rely on typeof this variable, but I suggest storing it inside AppLogo.tsx

Comment (ValKoDim) [reply code/src/components/app-logo/AppLogo.types.ts]: Okay, I moved it back to AppLogo.tsx

File: code/src/components/app-icon-button/AppIconButton.tsx (hunk 4 above)
Comment (ValKoDim) [reply code/src/components/app-icon-button/AppIconButton.tsx]: Done

File: code/src/components/app-button/AppButton.tsx (hunk 3 above)
Comment (ValKoDim) [reply code/src/components/app-button/AppButton.tsx]: Done

File: code/src/layouts/page-wrapper/PageWrapper.tsx (hunk 2 above)
Comment (ValKoDim) [reply code/src/layouts/page-wrapper/PageWrapper.tsx]: Done

**Review by mhevyk — APPROVED** [review]

File: code/src/components/app-container/AppContainer.tsx (hunk 1 above)
Comment (ValKoDim) [reply code/src/components/app-container/AppContainer.tsx]: Done

**Review by D1nviz — APPROVED** [review]

**Pull request merged**